modules/core/include/opencv2/core/hal/intrin_wasm.hpp

```diff
@@ -11,6 +11,10 @@
 #include <emscripten/version.h>
 #include "opencv2/core/saturate.hpp"
 
+#ifndef __EMSCRIPTEN_major__
+#include <emscripten/version.h>
```

Comment:
Line 11 do the same (and fail).

I triggered `javascript-simd` build configuration. It is based on EMSDK 2.0.13.

PR: https://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/100452
4.x (also fail): https://pullrequest.opencv.org/buildbot/builders/4_x_javascript-simd-emscripten-lin64/builds/100077

Reply:
@opencv-alalek also, my bad, it looks like the opencv version referenced by `vcpkg install opencv4:wasm32-emscripten`, `4.8.0-2bf495557d` is a different header inclusion.... I will rebase and retry. 

```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.

#ifndef OPENCV_HAL_INTRIN_WASM_HPP
#define OPENCV_HAL_INTRIN_WASM_HPP

#include <limits>
#include <cstring>
#include <algorithm>
#include "opencv2/core/saturate.hpp"

#ifndef __EMSCRIPTEN_major__
#include <emscripten/version.h>
#endif

#define CV_SIMD128 1
#define CV_SIMD128_64F 0 // Now all implementation of f64 use fallback, so disable it.
#define CV_SIMD128_FP16 0
```

Reply:
@opencv-alalek You'd mentioned 
> It is based on EMSDK 2.0.13.
So the same inclusion should not work anyways?...

Reply:
Apologies, missed line 11; opencv4.8.0 is the version for vcpkg port. 

> I triggered javascript-simd build configuration. It is based on EMSDK 2.0.13.
Given the current build arguments from  https://pullrequest.opencv.org/buildbot/builders/4_x_javascript-simd-emscripten-lin64/builds/100077 It looks like the breaking behavior is predictable, cause; 

```
Executing: ['cmake', '-DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3', '-DENABLE_PIC=FALSE', '-DCMAKE_BUILD_TYPE=Release', "-DCPU_BASELINE=''", '-DCMAKE_INSTALL_PREFIX=/usr/local', "-DCPU_DISPATCH=''", '-DCV_TRACE=OFF', '-DBUILD_SHARED_LIBS=OFF', '-DWITH_1394=OFF', '-DWITH_ADE=OFF', '-DWITH_VTK=OFF', '-DWITH_EIGEN=OFF', '-DWITH_FFMPEG=OFF', '-DWITH_GSTREAMER=OFF', '-DWITH_GTK=OFF', '-DWITH_GTK_2_X=OFF', '-DWITH_IPP=OFF', '-DWITH_JASPER=OFF', '-DWITH_JPEG=OFF', '-DWITH_WEBP=OFF', '-DWITH_OPENEXR=OFF', '-DWITH_OPENGL=OFF', '-DWITH_OPENVX=OFF', '-DWITH_OPENNI=OFF', '-DWITH_OPENNI2=OFF', '-DWITH_PNG=OFF', '-DWITH_TBB=OFF', '-DWITH_TIFF=OFF', '-DWITH_V4L=OFF', '-DWITH_OPENCL=OFF', '-DWITH_OPENCL_SVM=OFF', '-DWITH_OPENCLAMDFFT=OFF', '-DWITH_OPENCLAMDBLAS=OFF', '-DWITH_GPHOTO2=OFF', '-DWITH_LAPACK=OFF', '-DWITH_ITT=OFF', '-DWITH_QUIRC=OFF', '-DBUILD_ZLIB=ON', '-DBUILD_opencv_apps=OFF', '-DBUILD_opencv_calib3d=ON', '-DBUILD_opencv_dnn=ON', '-DBUILD_opencv_features2d=ON', '-DBUILD_opencv_flann=ON', '-DBUILD_opencv_gapi=OFF', '-DBUILD_opencv_ml=OFF', '-DBUILD_opencv_photo=ON', '-DBUILD_opencv_imgcodecs=OFF', '-DBUILD_opencv_shape=OFF', '-DBUILD_opencv_videoio=OFF', '-DBUILD_opencv_videostab=OFF', '-DBUILD_opencv_highgui=OFF', '-DBUILD_opencv_superres=OFF', '-DBUILD_opencv_stitching=OFF', '-DBUILD_opencv_java=OFF', '-DBUILD_opencv_js=ON', '-DBUILD_opencv_python2=OFF', '-DBUILD_opencv_python3=OFF', '-DBUILD_EXAMPLES=ON', '-DBUILD_PACKAGE=OFF', '-DBUILD_TESTS=ON', '-DBUILD_PERF_TESTS=ON', "-DCMAKE_TOOLCHAIN_FILE='/opt/emsdk-portable/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake'", '-DBUILD_DOCS=OFF', '-DWITH_PTHREADS_PF=OFF', '-DCV_ENABLE_INTRINSICS=ON', '-DBUILD_WASM_INTRIN_TESTS=ON', '-DCMAKE_C_FLAGS=\'-s WASM=1 -s SINGLE_FILE=1 -s USE_PTHREADS=0 -msimd128 -s EXPORTED_FUNCTIONS="[\'_malloc\', \'_free\']"\'', '-DCMAKE_CXX_FLAGS=\'-s WASM=1 -s SINGLE_FILE=1 -s USE_PTHREADS=0 -msimd128 -s EXPORTED_FUNCTIONS="[\'_malloc\', \'_free\']"\'', '/build/4_x_javascript-simd-emscripten-lin64/opencv']
```

doesn't include `__EMSCRIPTEN_major__`, etc. and OpenCV4.X uses Emscripte 2.0.13, which still relies on CLI def of the macros. Correct?